ironfish/src/rpc/routes/mempool/getTransactions.ts

```diff
@@ -0,0 +1,137 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { getFeeRate, MemPool } from '../../../memPool'
+import { Transaction } from '../../../primitives'
+import { ApiNamespace, router } from '../router'
+
+type MinMax = {
+  min?: number
+  max?: number
+}
+
+export type GetMempoolTransactionsRequest = {
+  limit?: number
+  feeRate?: MinMax
+  fee?: MinMax
+  expiration?: MinMax
+  position?: MinMax
+  stream?: boolean
+}
+
+export type GetMempoolTransactionResponse = {
+  serializedTransaction: string
+  position: number
+}
+
+export type GetMempoolTransactionsResponse = {
+  transactions: GetMempoolTransactionResponse[]
+}
+
+const minMaxSchema = yup.object({
+  min: yup.number().optional().min(0),
+  max: yup.number().optional(),
+})
+
+export const MempoolTransactionsRequestSchema: yup.ObjectSchema<GetMempoolTransactionsRequest> =
+  yup
+    .object({
+      limit: yup.number().min(0),
+      feeRate: minMaxSchema.optional(),
+      fee: minMaxSchema.optional(),
+      expiration: minMaxSchema.optional(),
+      position: minMaxSchema.optional(),
+      stream: yup.boolean().optional(),
+    })
+    .required()
+    .defined()
+
+export const MempoolTransactionsResponseSchema: yup.ObjectSchema<GetMempoolTransactionsResponse> =
+  yup
+    .object({
+      transactions: yup
+        .array()
+        .of(
+          yup
+            .object({
+              serializedTransaction: yup.string().defined(),
+              position: yup.number().defined(),
+            })
+            .defined(),
+        )
+        .defined(),
+    })
+    .defined()
+
+router.register<typeof MempoolTransactionsRequestSchema, GetMempoolTransactionsResponse>(
+  `${ApiNamespace.mempool}/getTransactions`,
+  MempoolTransactionsRequestSchema,
+  (request, node): void => {
+    if (!request.data.stream) {
+      request.end({ transactions: getTransactions(node.memPool, request.data) })
+      return
+    }
+
+    request.stream({
+      transactions: getTransactions(node.memPool, request.data),
+    })
+
+    const interval = setInterval(() => {
+      request.stream({
+        transactions: getTransactions(node.memPool, request.data),
+      })
+    }, 5000)
+
+    request.onClose.on(() => {
+      clearInterval(interval)
+    })
+
+    request.end({
+      transactions: getTransactions(node.memPool, request.data),
+    })
+  },
+)
+
+function getTransactions(memPool: MemPool, request: GetMempoolTransactionsRequest) {
+  const serializedTransactions: GetMempoolTransactionResponse[] = []
+
+  let position = 0
+  for (const transaction of memPool.orderedTransactions()) {
+    if (request.limit !== undefined && serializedTransactions.length >= request.limit) {
+      break
+    }
+
+    if (includeTransaction(transaction, position, request)) {
+      serializedTransactions.push({
```

Comment:
this can blow node memory with how large the mempool can get you should stream these back

Reply:
Ok so sounds like your suggesting is to just get rid of mempool streaming altogether (we only allow one time queries of the mempool). And those one time queries we would stream the results. I think that makes sense to me. Streaming the transaction list with small mempool was pretty cool to see transactions being added in real time but it's by no means necessary so I'm ok with getting rid of it